test/unit/utils_test.js

```diff
@@ -94,24 +150,34 @@ describe('Utils', function() {
       expect(maxTime).to.eql(600);
       expect(maxTime - minTime).to.eql(600);
     });
-  });
 
-  describe("Utils.readChallengeTx", function() {
-    it('requires a non-muxed account', function() {
+    it("throws an error if a muxed account and memo is passed", function () {
+      let keypair = StellarSdk.Keypair.random();
+      const muxedAccount = new StellarSdk.MuxedAccount(
+        new StellarSdk.Account(
+          StellarSdk.Keypair.random().publicKey(), "-1"
+        ),
+        "5842698851377328257"
+      )
+
       expect(() =>
-        StellarSdk.Utils.readChallengeTx(
-          "avalidtx",
-          "MAAAAAAAAAAAAAB7BQ2L7E5NBWMXDUCMZSIPOBKRDSBYVLMXGSSKF6YNPIB7Y77ITLVL6",
-          "SDF",
-          300,
+        StellarSdk.Utils.buildChallengeTx(
+          keypair,
+          muxedAccount.accountId(),
```

Comment:
:question: Why not just use the M-address? I don't see `muxedAccount` being used elsewhere.

Reply:
I just favored not hard-coding the `M` address, although I do that in other tests so I'll be consistent and use hard-coded M addresses 👍